tests/Feature/Users/ProfileTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Events\UserUpdated;
+use App\User;
+
+use DB;
+use Carbon\Carbon;
+use Illuminate\Auth\AuthenticationException;
+use Illuminate\Validation\UnauthorizedException;
+use Tests\TestCase;
+use Illuminate\Support\Facades\Event;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class ProfileTest extends TestCase
+{
+    public function testProfilePage() {
+        $user = factory(User::class)->states('Restarter')->create();
+
+        // When logged out should throw an exception.
+        try {
+            $response = $this->get('/profile');
+            $this->assertFalse(TRUE);
+        } catch (AuthenticationException $e) {
+            // Success case.
+        }
+
+        // When logged in should be able to see.
+        // TODO I'm not convinced that viewing /profile is ever reachable, though /profile/id is.
```

Comment:
/profile with no id shows you your own profile, I believe.

Reply:
But do we link to it anywhere?

Reply:
Ah, right - at present, I don't know actually!

The only thing I recall related to this, was allowing for the ability to link from elsewhere, e.g. Talk, or the Wiki, to the user's profile page, without requiring knowing their id in Restarters.